src/runtime/internal/transitions.ts

```diff
@@ -79,6 +79,9 @@ export function transition_out(block: Fragment, local: 0 | 1, detach?: 0 | 1, ca
 		});
 
 		block.o(local);
+		// fix: destroy non-fragment element such as empty components.
+	} else if (typeof callback === 'function') {
```

Comment:
According to actual usecases, type of `callback` is `callback?: () => void`.
So we should declare the type at line 68.
And just we need `} else if (callback) {`.

Reply:
Still hold my point, as `callback?: () => void` is validated in compilation stage by typescript, but `typeof callback === 'function' ` is validated in runtime stage. Considering this callback is passed by user, it would be better to do strict check here.

Reply:
> Considering this callback is passed by user,

I recognized that `user` means Svelte users.
If it's correct, `transition_out` is not a public function from the point of view of Svelte users.
Only the compiler can use it.

Reply:
> > Considering this callback is passed by user,
> 
> I recognized that `user` means Svelte users. If it's correct, `transition_out` is not a public function from the point of view of Svelte users. Only the compiler can use it.

Got it. After carefully viewing the caller, the callback can only be called by inner. 
![Screen Shot 2022-05-06 at 11 36 22 AM](https://user-images.githubusercontent.com/30370223/167062857-e4b9f421-7fe2-4454-8c0d-0b4a411e907a.png)